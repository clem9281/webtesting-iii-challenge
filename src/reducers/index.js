import {TOGGLE_LOCK, TOGGLE_CLOSE} from "../actions";

export const initialState = {
    locked: false,
    closed: false
}

export const reducer = (state = initialState, action) => {
    switch(action.type) {
        case TOGGLE_LOCK:
            return {
                ...state,
                locked: !state.locked
            }
        case TOGGLE_CLOSE:
            return {
                ...state,
                closed: !state.closed
            }
        default: 
            return {
                ...state
            }
    }
}