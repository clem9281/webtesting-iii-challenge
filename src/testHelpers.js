import React from 'react';
import {Provider} from "react-redux";
import {createStore} from "redux";
import { render} from "react-testing-library";

import { reducer } from "./reducers";

export const closedStore = {
    closed: true
}

export const openStore = {
    closed: false
}

export const lockedStore = {
    locked: true
}

export const unlockedStore = {
    locked: false
}

export const renderWithRedux = (ui, state) => {
    const store = createStore(reducer, state);
    return {
        ...render(<Provider store={store}>{ui}</Provider>),
      store,
    }
}