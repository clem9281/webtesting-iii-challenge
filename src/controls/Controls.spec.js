// Test away!
import React from "react";
import { render, cleanup } from "react-testing-library";
import renderer from "react-test-renderer";
import "jest-dom/extend-expect";
import {Provider} from "react-redux";
import {createStore} from "redux";

import { reducer } from "../reducers";

import Controls from "./Controls";

import {renderWithRedux, openStore, lockedStore} from "../testHelpers";

afterEach(cleanup);

describe('Controls', () => {
    it("should match the snapshot", () => {
        const tree = renderer.create(<Provider store={createStore(reducer)}><Controls /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("should have button to toggle the closed and locked states", () => {
        const { getByText } = renderWithRedux(<Controls />);
        getByText(/lock gate/i);
        getByText(/close gate/i);
    })

    it("should display the lock button as disabled if the gate is open", () => {
        const { getByText } = renderWithRedux(<Controls/>, openStore);
        const lockBtn = getByText(/lock gate/i);
        expect(lockBtn).toBeDisabled();
    })

    it("should display the close button as disabled if the gate is locked", () => {
        const { getByText } = renderWithRedux(<Controls/>, lockedStore);
        const closeBtn = getByText(/close gate/i);
        expect(closeBtn).toBeDisabled();
    })
})