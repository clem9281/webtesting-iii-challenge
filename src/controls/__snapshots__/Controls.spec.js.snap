// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match the snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
    onClick={[Function]}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    disabled={false}
    onClick={[Function]}
  >
    Close Gate
  </button>
</div>
`;
